### PR TITLE
(dbt) Add intermediate user iris mapping model

### PR DESCRIPTION
**Pull request by carolinemestre-passculture**
# DE PR

## Describe your changes

This PR is linked to[ this ticket](https://www.notion.so/passcultureapp/Add-intermediate-user-geo_iris-models-03b3bffa0c854ff792f6f37316a5c767) : 
We want to add an intermediate model : `int_addresses_user_iris` that calculates the associated iris_id for each user. We set this model to incremental so that it only calculates the `iris_id` for users who have been updated each day.

Tag a reviewer if necessacy  @cdelabre 

## Jira ticket number and/or notion link

JIRA-ticket_number

### Type of change
- [ ] Fix (non-breaking change which corrects expected behavior)
- [ ] New fields (non-breaking change)
- [ ] New table (non-breaking change)
- [ ] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts) 
- [ ] Table deletion (potentially breaking change which adds functionality/ table)
      
### Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] My code passes CI/CD tests
- [ ] I updated README.md
- [ ] I have updated the dag
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?
- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] ⏰ no, but I created a ticket

### CODE REVIEW

File: orchestration/dags/data_gcp_dbt/models/intermediate/addresses/int_addresses__user_geo_iris.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        on_schema_change = "sync_all_columns",
+    )
+}}
+
+WITH users_updated AS (
+SELECT user_id,
+    latitude AS user_latitude,
+    longitude AS user_longitude,
+    date_updated
+FROM {{ source("analytics", "user_locations") }}
```
Comment (cdelabre) [inline orchestration/dags/data_gcp_dbt/models/intermediate/addresses/int_addresses__user_geo_iris.sql:15]: Je vois trois sources `user_locations` (clean, raw, analytics)`. Est-ce que l'on peut utiliser les sources raw ou clean plutôt afin de supprimer par la suite la partie analytics ?

Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/addresses/int_addresses__user_geo_iris.sql:15]: Yes, on a utilisé celle de analytics car est dé dupliquée.

File: orchestration/dags/data_gcp_dbt/models/intermediate/addresses/int_addresses__user_geo_iris.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{{
```
Comment (cdelabre) [inline orchestration/dags/data_gcp_dbt/models/intermediate/addresses/int_addresses__user_geo_iris.sql:1]: api_gouv/addreses/int_api_gouv__user_geo_iris.sql

File: orchestration/dags/data_gcp_dbt/models/intermediate/api_gouv/addresses/int_api_gouv__user_address_location.sql
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/intermediate/api_gouv/addresses/int_api_gouv__user_address_location.sql:1]: @cdelabre ou int_api_gouv__address_user_location (pour commencer par le nom du sous sous dossier la 2nde partie)

File: orchestration/dags/data_gcp_dbt/models/intermediate/api_gouv/addresses/int_api_gouv__user_geo_iris.sql
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/intermediate/api_gouv/addresses/int_api_gouv__user_geo_iris.sql:1]: pareil ici mettre address dans le nom : int_api_gouv__address_user_geo_iris

**Review by cdelabre — DISMISSED** [review]

**Review by cdelabre — APPROVED** [review]

**Pull request merged**